### PR TITLE
Fix Asset Hub collator crashing when starting from genesis

### LINKED ISSUE #1566: Error after upgrading polkadot-parachain to v1.1.0-polkadot on Asset Hub
I've updated 6 system chains with ansible-galaxy. From v1.0.0 to v1.1.0, I'm using the binary that is published in the releases.  5 out of 6 upgrades went smooth. One of the machines is giving an error which I can't solve. 

```
2023-09-14 13:25:32.761  INFO main sc_cli::runner: Polkadot parachain
2023-09-14 13:25:32.761  INFO main sc_cli::runner: ✌️  version 1.1.0-783579b4a49
2023-09-14 13:25:32.761  INFO main sc_cli::runner: ❤️  by Parity Technologies <admin@parity.io>, 2017-2023
2023-09-14 13:25:32.761  INFO main sc_cli::runner: 📋 Chain specification: Polkadot Asset Hub
....
....
2023-09-14 13:13:14.603  WARN tokio-runtime-worker sc_service::builder: [Parachain] The NetworkStart returned as part of `build_network` has been silently dropped
Error: Service(Client(RuntimeApiError(Application(Execution(Other("Exported method AuraApi_slot_duration is not found"))))))
asset-hub-dot.service: Main process exited, code=exited, status=1/FAILURE
asset-hub-dot.service: Failed with result 'exit-code'.
```

There are no other warnings or errors in the start-up logs after the starting. After that it just keeps rebooting.

I have deleted the DB, since it was a redundant node and tried to re-sync it, but the error pertains. 

The exec parameters
```
COMMON="\
--base-path /home/statemint/.local/share/polkadot \
--detailed-log-output"
RC_CHAIN="--chain polkadot"
RC_ADDR="\
--listen-addr=/ip4/0.0.0.0/tcp/30338 \
--public-addr=/ip4/xxxx/tcp/30338"
RC_CONNECTIONS="--in-peers 25 --out-peers 25"
RC_DB="\
--database paritydb \"
RC_PRUNING="--state-pruning=256"
RC_METRICS="\
--prometheus-external \
--prometheus-port 9620"
RC_RPC="--rpc-port 10000"


PC_NAME="--name 'Staker Space [2]'"
PC_ROLE_SPECIFIC="\
--collator"
PC_CHAIN="--chain asset-hub-polkadot"
PC_REMOTE_RC_URLS=""
PC_ADDR="\
--listen-addr=/ip4/0.0.0.0/tcp/31555 \
--public-addr=/ip4/xxxx/tcp/31555"
PC_CONNECTIONS="--in-peers 25 --out-peers 25"
PC_DB="\--database paritydb \"
PC_PRUNING="--state-pruning=256"
PC_LOGS=""
PC_METRICS="\
--prometheus-external \
--prometheus-port 9621"
PC_WS="\
--rpc-max-connections 100"
PC_RPC="--rpc-port 10001"
```

The server:

```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
kernel: 5.4.0-144-generic
cpu :Core(TM) i7-9700K CPU @ 3.60GHz
```

### PR DESCRIPTION
**Pull request by georgepisaltu**
Fixes #1566 

# Description

When running `polkadot-parachain --chain asset-hub-*` with no DB, the node would crash because it would try to access Aura API which wasn't available in the genesis `shell` runtime that Asset Hub started with. More details on the issue [here](https://github.com/paritytech/polkadot-sdk/issues/1566#issuecomment-1740649027).

This PR makes the node start with a relay chain driven consensus which doesn't call into `AuraApi::slot_duration`, then after the API becomes available (after the runtime is upgraded to the proper Asset Hub), it uses the Aura consensus.

# Checklist

- [ ] My PR includes a detailed description as outlined in the "Description" section above
- [ ] My PR follows the [labeling requirements](CONTRIBUTING.md#Process) of this project (at minimum one label for `T`
  required)
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)

### CODE REVIEW

**Review by slumber — APPROVED** [review]

**Review by BradleyOlson64 — APPROVED** [review]

File: cumulus/client/collator/src/lib.rs [hunk 1]
```diff
@@ -69,14 +69,14 @@ where
 	RA::Api: CollectCollationInfo<Block>,
 {
 	/// Create a new instance.
-	fn new(
+	pub fn new(
```
Comment (bkchr) [inline cumulus/client/collator/src/lib.rs:72]: Actually revert all the changes in this file.

File: cumulus/polkadot-parachain/src/service.rs [hunk 2]
```diff
@@ -1389,6 +1390,188 @@ where
...
+						collator
+							.clone()
+							.produce_candidate(
+								*request.relay_parent(),
+								request.persisted_validation_data().clone(),
+							)
+							.await
+					};
+
+					request.complete(collation);
```
[the remark this answers is not in the document]
Comment (bkchr) [reply cumulus/polkadot-parachain/src/service.rs:1537]: We actually don't need all of this code. Just decode the header manually and then check if the runtime api is there.

**Review by bkchr — CHANGES_REQUESTED** [review]
Generally looking good, but some of the details need to improve.

File: cumulus/polkadot-parachain/src/service.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgepisaltu) [reply cumulus/polkadot-parachain/src/service.rs:1537]: Done.

File: cumulus/client/collator/src/lib.rs (hunk 1 above)
Comment (georgepisaltu) [reply cumulus/client/collator/src/lib.rs:72]: Done.

File: cumulus/polkadot-parachain/src/service.rs [hunk 3]
```diff
@@ -1389,6 +1392,150 @@ where
...
+								continue
+							},
+						};
+
+					let last_head_hash = last_header.hash();
+
+					if !collator_service.check_block_status(last_head_hash, &last_header) {
+						request.complete(None);
+						continue
+					}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/src/service.rs:1487]:
```suggestion
```

No need to do this.

File: cumulus/polkadot-parachain/src/service.rs [hunk 4]
```diff
@@ -1389,6 +1392,150 @@ where
...
+						.unwrap_or(false)
+					{
+						// Respond to this request before transitioning to Aura.
+						request.complete(None);
+						break
+					}
+				}
+
+				// Move to Aura consensus.
+				let slot_duration = cumulus_client_consensus_aura::slot_duration(&*client).unwrap();
```
Comment (bkchr) [inline cumulus/polkadot-parachain/src/service.rs:1503]: No unwrap. Please print the error and return.

File: cumulus/polkadot-parachain/src/service.rs [hunk 5]
```diff
@@ -1389,6 +1392,150 @@ where
...
+					// Very limited proposal time.
+					authoring_duration: Duration::from_millis(500),
+				};
+
+				basic_aura::run::<Block, <AuraId as AppCrypto>::Pair, _, _, _, _, _, _, _>(params)
+					.await
+			});
+
+			let spawner = task_manager.spawn_handle();
+			spawner.spawn("cumulus-asset-hub-collator", None, collation_future);
```
Comment (bkchr) [inline cumulus/polkadot-parachain/src/service.rs:1530]:
```suggestion
			spawner.spawn_essential("cumulus-asset-hub-collator", None, collation_future);
```

File: cumulus/polkadot-parachain/src/service.rs [hunk 6]
```diff
@@ -1389,6 +1392,150 @@ where
...
+					collator_service,
+					// Very limited proposal time.
+					authoring_duration: Duration::from_millis(500),
+				};
+
+				basic_aura::run::<Block, <AuraId as AppCrypto>::Pair, _, _, _, _, _, _, _>(params)
+					.await
+			});
+
+			let spawner = task_manager.spawn_handle();
```
Comment (bkchr) [inline cumulus/polkadot-parachain/src/service.rs:1529]:
```suggestion
			let spawner = task_manager.spawn_essential_handle();
```

**Review by bkchr — APPROVED** [review]
Please fix the last issues and then you can merge it! Ty!

File: cumulus/polkadot-parachain/src/service.rs (hunk 4 above)
Comment (georgepisaltu) [reply cumulus/polkadot-parachain/src/service.rs:1503]: Good catch, I forgot about this.

**Pull request merged**